File: cscctest/csharp/lexer/literal1.cs
/*
 * literal1.cs - Test the handling of valid literals in the lexer.
 *
 * "C# Language Specification", Draft 13, Section 9.4.4, "Literals"
 *
 * Copyright (C) 2002  Southern Storm Software, Pty Ltd.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 2 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program; if not, write to the Free Software
 * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
 */

class Test
{
	// Boolean literals.
	public const bool True  = true;
	public const bool False = false;

	// Integer literals.
	public const int i1 = 0;
	public const int i2 = 012;		// Octal warning, recognised as decimal.
	public const int i3 = 123456;
	public const int i4 = -123456;
	public const int i5 = 0x123456;
	public const int i6 = 0x7FFFFFFF;
	public const int i7 = 0x7abcdef0;
	public const int i8 = 2147483647;
	public const int i9 = -2147483648;

	// Unsigned integer literals.
	public const uint ui1 = 0;
	public const uint ui2 = 123456;
	public const uint ui3 = 123456u;
	public const uint ui4 = 0x123456U;
	public const uint ui5 = 2147483647;
	public const uint ui6 = 2147483648;
	public const uint ui7 = 0xFFFFFFFF;

	// Long integer literals.
	public const long l1 = 0;
	public const long l2 = 123456;
	public const long l3 = 2147483647;
	public const long l4 = 2147483648;
	public const long l5 = -2147483648;
	public const long l6 = 0xFFFFFFFF;
	public const long l7 = 0x100000000;
	public const long l8 = -0x100000000;
	public const long l9 = 123456l;
	public const long l10 = 123456L;
	public const long l11 = 9223372036854775807;
	public const long l12 = -9223372036854775808;

	// Unsigned long integer literals.
	public const ulong ul1 = 0;
	public const ulong ul2 = 123456;
	public const ulong ul3 = 2147483647;
	public const ulong ul4 = 2147483648;
	public const ulong ul5 = 0xFFFFFFFF;
	public const ulong ul6 = 0x100000000;
	public const ulong ul7 = 123456l;
	public const ulong ul8 = 123456uL;
	public const ulong ul9 = 9223372036854775807;
	public const ulong ul10 = 9223372036854775808;
	public const ulong ul11 = 0xFFFFFFFFFFFFFFFF;
	public const ulong ul12 = 123456u;
	public const ulong ul13 = 123456ul;
	public const ulong ul14 = 123456uL;
	public const ulong ul15 = 123456Ul;
	public const ulong ul16 = 123456UL;
	public const ulong ul17 = 123456lu;
	public const ulong ul18 = 123456lU;
	public const ulong ul19 = 123456Lu;
	public const ulong ul20 = 123456LU;

	// Single-precision floating-point literals.
	public const float f1 = 0.0f;
	public const float f2 = -1.0F;
	public const float f3 = 1234.56e23f;
	public const float f4 = 1234.56e+23F;
	public const float f5 = 1234.56e-23f;
	public const float f6 = .123F;
	public const float f7 = .123e23f;
	public const float f8 = 123e23F;
	public const float f9 = 123e+23f;
	public const float f10 = 123e-23F;
	public const float f11 = 123f;

	// Double-precision floating-point literals.
	public const double d1 = 0.0d;
	public const double d2 = -1.0D;
	public const double d3 = 1234.56e23;
	public const double d4 = 1234.56e+23d;
	public const double d5 = 1234.56e-23D;
	public const double d6 = .123;
	public const double d7 = .123e23d;
	public const double d8 = 123e23D;
	public const double d9 = 123e+23;
	public const double d10 = 123e-23d;
	public const double d11 = 123D;

	// Decimal literals.
	public const decimal dc1 = 0.0m;
	public const decimal dc2 = -1.0M;
	public const decimal dc3 = 1234.56e23m;
	public const decimal dc4 = 1234.56e+23M;
	public const decimal dc5 = 1234.56e-23m;
	public const decimal dc6 = .123M;
	public const decimal dc7 = .123e23m;
	public const decimal dc8 = 123e23M;
	public const decimal dc9 = 123e+23m;
	public const decimal dc10 = 123e-23M;
	public const decimal dc11 = 123m;
	public const decimal dc12 = 79228162514264337593543950335m;
	public const decimal dc13 = -79228162514264337593543950335m;
	public const decimal dc14 = -792281625.14264337593543950335m;
	public const decimal dc15 = -7.9228162514264337593543950335m;
	public const decimal dc16 = -0.79228162514264337593543950335m;
	public const decimal dc17 = 0.0000000000000000000000000001m;
	public const decimal dc18 = 0.00000000000000000000000000001m; // == 0.0m

	// Character literals.
	public const char c1 = 'a';
	public const char c2 = '\'';
	public const char c3 = '\"';
	public const char c4 = '\\';
	public const char c5 = '\0';
	public const char c6 = '\a';
	public const char c7 = '\b';
	public const char c8 = '\f';
	public const char c9 = '\n';
	public const char c10 = '\r';
	public const char c11 = '\t';
	public const char c12 = '\v';
	public const char c13 = '\x3';
	public const char c14 = '\x03';
	public const char c15 = '\x003';
	public const char c16 = '\x0003';
	public const char c17 = '\u0003';
	public const char c18 = '\U00000003';
	public const char c19 = '\033';		// Octal character warning.

	// String literals.
	public const string s1 = null;
	public const string s2 = "";
	public const string s3 = "Hello World!";
	public const string s4 = @"Hello \tWo""rld!";
	public const string s5 = @"line 1
line 2
line 3";
	public const string s6 = @"
#bad
";
	public const string s7 = "\U00101234";

	// Object literals.
	public const object o1 = null;
	public const Test   o2 = null;
}

File: cscctest/csharp/lexer/literal1.jl
.assembly extern '.library'
{
	.ver 0:0:0:0
}
.assembly '<Assembly>'
{
	.ver 0:0:0:0
}
.module '<Module>'
.class private auto ansi beforefieldinit 'Test' extends ['.library']'System'.'Object'
{
.field public static literal bool 'True' = bool(true)
.field public static literal bool 'False' = bool(false)
.field public static literal int32 'i1' = int32(0x00000000)
.field public static literal int32 'i2' = int32(0x0000000C)
.field public static literal int32 'i3' = int32(0x0001E240)
.field public static literal int32 'i4' = int32(0xFFFE1DC0)
.field public static literal int32 'i5' = int32(0x00123456)
.field public static literal int32 'i6' = int32(0x7FFFFFFF)
.field public static literal int32 'i7' = int32(0x7ABCDEF0)
.field public static literal int32 'i8' = int32(0x7FFFFFFF)
.field public static literal int32 'i9' = int32(0x80000000)
.field public static literal unsigned int32 'ui1' = int32(0x00000000)
.field public static literal unsigned int32 'ui2' = int32(0x0001E240)
.field public static literal unsigned int32 'ui3' = int32(0x0001E240)
.field public static literal unsigned int32 'ui4' = int32(0x00123456)
.field public static literal unsigned int32 'ui5' = int32(0x7FFFFFFF)
.field public static literal unsigned int32 'ui6' = int32(0x80000000)
.field public static literal unsigned int32 'ui7' = int32(0xFFFFFFFF)
.field public static literal int64 'l1' = int64(0x0000000000000000)
.field public static literal int64 'l2' = int64(0x000000000001E240)
.field public static literal int64 'l3' = int64(0x000000007FFFFFFF)
.field public static literal int64 'l4' = int64(0x0000000080000000)
.field public static literal int64 'l5' = int64(0xFFFFFFFF80000000)
.field public static literal int64 'l6' = int64(0x00000000FFFFFFFF)
.field public static literal int64 'l7' = int64(0x0000000100000000)
.field public static literal int64 'l8' = int64(0xFFFFFFFF00000000)
.field public static literal int64 'l9' = int64(0x000000000001E240)
.field public static literal int64 'l10' = int64(0x000000000001E240)
.field public static literal int64 'l11' = int64(0x7FFFFFFFFFFFFFFF)
.field public static literal int64 'l12' = int64(0x8000000000000000)
.field public static literal unsigned int64 'ul1' = int64(0x0000000000000000)
.field public static literal unsigned int64 'ul2' = int64(0x000000000001E240)
.field public static literal unsigned int64 'ul3' = int64(0x000000007FFFFFFF)
.field public static literal unsigned int64 'ul4' = int64(0x0000000080000000)
.field public static literal unsigned int64 'ul5' = int64(0x00000000FFFFFFFF)
.field public static literal unsigned int64 'ul6' = int64(0x0000000100000000)
.field public static literal unsigned int64 'ul7' = int64(0x000000000001E240)
.field public static literal unsigned int64 'ul8' = int64(0x000000000001E240)
.field public static literal unsigned int64 'ul9' = int64(0x7FFFFFFFFFFFFFFF)
.field public static literal unsigned int64 'ul10' = int64(0x8000000000000000)
.field public static literal unsigned int64 'ul11' = int64(0xFFFFFFFFFFFFFFFF)
.field public static literal unsigned int64 'ul12' = int64(0x000000000001E240)
.field public static literal unsigned int64 'ul13' = int64(0x000000000001E240)
.field public static literal unsigned int64 'ul14' = int64(0x000000000001E240)
.field public static literal unsigned int64 'ul15' = int64(0x000000000001E240)
.field public static literal unsigned int64 'ul16' = int64(0x000000000001E240)
.field public static literal unsigned int64 'ul17' = int64(0x000000000001E240)
.field public static literal unsigned int64 'ul18' = int64(0x000000000001E240)
.field public static literal unsigned int64 'ul19' = int64(0x000000000001E240)
.field public static literal unsigned int64 'ul20' = int64(0x000000000001E240)
.field public static literal float32 'f1' = float32(0x00000000)
.field public static literal float32 'f2' = float32(0xBF800000)
.field public static literal float32 'f3' = float32(0x6ACC3DA6)
.field public static literal float32 'f4' = float32(0x6ACC3DA6)
.field public static literal float32 'f5' = float32(0x1E6933A9)
.field public static literal float32 'f6' = float32(0x3DFBE76D)
.field public static literal float32 'f7' = float32(0x6426B233)
.field public static literal float32 'f8' = float32(0x6922CA05)
.field public static literal float32 'f9' = float32(0x6922CA05)
.field public static literal float32 'f10' = float32(0x1CB9DF52)
.field public static literal float32 'f11' = float32(0x42F60000)
.field public static literal float64 'd1' = float64(0x0000000000000000)
.field public static literal float64 'd2' = float64(0xBFF0000000000000)
.field public static literal float64 'd3' = float64(0x455987B4CAE67FA8)
.field public static literal float64 'd4' = float64(0x455987B4CAE67FA8)
.field public static literal float64 'd5' = float64(0x3BCD267511916BC9)
.field public static literal float64 'd6' = float64(0x3FBF7CED916872B0)
.field public static literal float64 'd7' = float64(0x4484D64651FE74C6)
.field public static literal float64 'd8' = float64(0x45245940AC127E09)
.field public static literal float64 'd9' = float64(0x45245940AC127E09)
.field public static literal float64 'd10' = float64(0x3B973BEA3F91F75A)
.field public static literal float64 'd11' = float64(0x405EC00000000000)
.field public static initonly valuetype ['.library']'System'.'Decimal' 'dc1'
.custom instance void ['.library']'System.Runtime.CompilerServices'.'DecimalConstantAttribute'::'.ctor'(unsigned int8, unsigned int8, unsigned int32, unsigned int32, unsigned int32) = (01 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00)
.field public static initonly valuetype ['.library']'System'.'Decimal' 'dc2'
.custom instance void ['.library']'System.Runtime.CompilerServices'.'DecimalConstantAttribute'::'.ctor'(unsigned int8, unsigned int8, unsigned int32, unsigned int32, unsigned int32) = (01 00 00 80 00 00 00 00 00 00 00 00 01 00 00 00 00 00)
.field public static initonly valuetype ['.library']'System'.'Decimal' 'dc3'
.custom instance void ['.library']'System.Runtime.CompilerServices'.'DecimalConstantAttribute'::'.ctor'(unsigned int8, unsigned int8, unsigned int32, unsigned int32, unsigned int32) = (01 00 00 00 D3 1E 66 00 A0 FE 99 2B 00 00 00 E8 00 00)
.field public static initonly valuetype ['.library']'System'.'Decimal' 'dc4'
.custom instance void ['.library']'System.Runtime.CompilerServices'.'DecimalConstantAttribute'::'.ctor'(unsigned int8, unsigned int8, unsigned int32, unsigned int32, unsigned int32) = (01 00 00 00 D3 1E 66 00 A0 FE 99 2B 00 00 00 E8 00 00)
.field public static initonly valuetype ['.library']'System'.'Decimal' 'dc5'
.custom instance void ['.library']'System.Runtime.CompilerServices'.'DecimalConstantAttribute'::'.ctor'(unsigned int8, unsigned int8, unsigned int32, unsigned int32, unsigned int32) = (01 00 19 00 00 00 00 00 00 00 00 00 40 E2 01 00 00 00)
.field public static initonly valuetype ['.library']'System'.'Decimal' 'dc6'
.custom instance void ['.library']'System.Runtime.CompilerServices'.'DecimalConstantAttribute'::'.ctor'(unsigned int8, unsigned int8, unsigned int32, unsigned int32, unsigned int32) = (01 00 03 00 00 00 00 00 00 00 00 00 7B 00 00 00 00 00)
.field public static initonly valuetype ['.library']'System'.'Decimal' 'dc7'
.custom instance void ['.library']'System.Runtime.CompilerServices'.'DecimalConstantAttribute'::'.ctor'(unsigned int8, unsigned int8, unsigned int32, unsigned int32, unsigned int32) = (01 00 00 00 9A 02 00 00 CE 3F CA C8 00 00 B0 98 00 00)
.field public static initonly valuetype ['.library']'System'.'Decimal' 'dc8'
.custom instance void ['.library']'System.Runtime.CompilerServices'.'DecimalConstantAttribute'::'.ctor'(unsigned int8, unsigned int8, unsigned int32, unsigned int32, unsigned int32) = (01 00 00 00 A0 2C 0A 00 04 3F 09 56 00 00 80 6F 00 00)
.field public static initonly valuetype ['.library']'System'.'Decimal' 'dc9'
.custom instance void ['.library']'System.Runtime.CompilerServices'.'DecimalConstantAttribute'::'.ctor'(unsigned int8, unsigned int8, unsigned int32, unsigned int32, unsigned int32) = (01 00 00 00 A0 2C 0A 00 04 3F 09 56 00 00 80 6F 00 00)
.field public static initonly valuetype ['.library']'System'.'Decimal' 'dc10'
.custom instance void ['.library']'System.Runtime.CompilerServices'.'DecimalConstantAttribute'::'.ctor'(unsigned int8, unsigned int8, unsigned int32, unsigned int32, unsigned int32) = (01 00 17 00 00 00 00 00 00 00 00 00 7B 00 00 00 00 00)
.field public static initonly valuetype ['.library']'System'.'Decimal' 'dc11'
.custom instance void ['.library']'System.Runtime.CompilerServices'.'DecimalConstantAttribute'::'.ctor'(unsigned int8, unsigned int8, unsigned int32, unsigned int32, unsigned int32) = (01 00 00 00 00 00 00 00 00 00 00 00 7B 00 00 00 00 00)
.field public static initonly valuetype ['.library']'System'.'Decimal' 'dc12'
.custom instance void ['.library']'System.Runtime.CompilerServices'.'DecimalConstantAttribute'::'.ctor'(unsigned int8, unsigned int8, unsigned int32, unsigned int32, unsigned int32) = (01 00 00 00 FF FF FF FF FF FF FF FF FF FF FF FF 00 00)
.field public static initonly valuetype ['.library']'System'.'Decimal' 'dc13'
.custom instance void ['.library']'System.Runtime.CompilerServices'.'DecimalConstantAttribute'::'.ctor'(unsigned int8, unsigned int8, unsigned int32, unsigned int32, unsigned int32) = (01 00 00 80 FF FF FF FF FF FF FF FF FF FF FF FF 00 00)
.field public static initonly valuetype ['.library']'System'.'Decimal' 'dc14'
.custom instance void ['.library']'System.Runtime.CompilerServices'.'DecimalConstantAttribute'::'.ctor'(unsigned int8, unsigned int8, unsigned int32, unsigned int32, unsigned int32) = (01 00 14 80 FF FF FF FF FF FF FF FF FF FF FF FF 00 00)
.field public static initonly valuetype ['.library']'System'.'Decimal' 'dc15'
.custom instance void ['.library']'System.Runtime.CompilerServices'.'DecimalConstantAttribute'::'.ctor'(unsigned int8, unsigned int8, unsigned int32, unsigned int32, unsigned int32) = (01 00 1C 80 FF FF FF FF FF FF FF FF FF FF FF FF 00 00)
.field public static initonly valuetype ['.library']'System'.'Decimal' 'dc16'
.custom instance void ['.library']'System.Runtime.CompilerServices'.'DecimalConstantAttribute'::'.ctor'(unsigned int8, unsigned int8, unsigned int32, unsigned int32, unsigned int32) = (01 00 1C 80 99 99 99 19 99 99 99 99 9A 99 99 99 00 00)
.field public static initonly valuetype ['.library']'System'.'Decimal' 'dc17'
.custom instance void ['.library']'System.Runtime.CompilerServices'.'DecimalConstantAttribute'::'.ctor'(unsigned int8, unsigned int8, unsigned int32, unsigned int32, unsigned int32) = (01 00 1C 00 00 00 00 00 00 00 00 00 01 00 00 00 00 00)
.field public static initonly valuetype ['.library']'System'.'Decimal' 'dc18'
.custom instance void ['.library']'System.Runtime.CompilerServices'.'DecimalConstantAttribute'::'.ctor'(unsigned int8, unsigned int8, unsigned int32, unsigned int32, unsigned int32) = (01 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00)
.field public static literal char 'c1' = char(0x0061)
.field public static literal char 'c2' = char(0x0027)
.field public static literal char 'c3' = char(0x0022)
.field public static literal char 'c4' = char(0x005C)
.field public static literal char 'c5' = char(0x0000)
.field public static literal char 'c6' = char(0x0007)
.field public static literal char 'c7' = char(0x0008)
.field public static literal char 'c8' = char(0x000C)
.field public static literal char 'c9' = char(0x000A)
.field public static literal char 'c10' = char(0x000D)
.field public static literal char 'c11' = char(0x0009)
.field public static literal char 'c12' = char(0x000B)
.field public static literal char 'c13' = char(0x0003)
.field public static literal char 'c14' = char(0x0003)
.field public static literal char 'c15' = char(0x0003)
.field public static literal char 'c16' = char(0x0003)
.field public static literal char 'c17' = char(0x0003)
.field public static literal char 'c18' = char(0x0003)
.field public static literal char 'c19' = char(0x001B)
.field public static literal class ['.library']'System'.'String' 's1' = nullref
.field public static literal class ['.library']'System'.'String' 's2' = ""
.field public static literal class ['.library']'System'.'String' 's3' = "Hello World!"
.field public static literal class ['.library']'System'.'String' 's4' = "Hello \\tWo\"rld!"
.field public static literal class ['.library']'System'.'String' 's5' = "line 1\nline 2\nline 3"
.field public static literal class ['.library']'System'.'String' 's6' = "\n#bad\n"
.field public static literal class ['.library']'System'.'String' 's7' = "\uDBC4\uDE34"
.field public static literal class ['.library']'System'.'Object' 'o1' = nullref
.field public static literal class 'Test' 'o2' = nullref
.method public hidebysig specialname rtspecialname instance void '.ctor'() cil managed java 
{
	aload_0
	invokespecial	instance void ['.library']'System'.'Object'::'.ctor'()
	return
	.locals 1
	.maxstack 1
} // method .ctor
.method private static hidebysig specialname rtspecialname void '.cctor'() cil managed java 
{
	iconst_0
	invokestatic	"System/Decimal" "op_Implicit__iV" "(I)LSystem/Decimal;"
	putstatic	valuetype ['.library']'System'.'Decimal' 'Test'::'dc1'
	iconst_m1
	invokestatic	"System/Decimal" "op_Implicit__iV" "(I)LSystem/Decimal;"
	putstatic	valuetype ['.library']'System'.'Decimal' 'Test'::'dc2'
	new	"System/Decimal"
	dup
	ldc	int32(-402653184)
	ldc	int32(731512480)
	ldc	int32(6692563)
	iconst_0
	iconst_0
	invokespecial	"System/Decimal" "<init>__iiiB" "(IIIZI)"
	putstatic	valuetype ['.library']'System'.'Decimal' 'Test'::'dc3'
	new	"System/Decimal"
	dup
	ldc	int32(-402653184)
	ldc	int32(731512480)
	ldc	int32(6692563)
	iconst_0
	iconst_0
	invokespecial	"System/Decimal" "<init>__iiiB" "(IIIZI)"
	putstatic	valuetype ['.library']'System'.'Decimal' 'Test'::'dc4'
	new	"System/Decimal"
	dup
	ldc	int32(123456)
	iconst_0
	iconst_0
	iconst_0
	bipush	25
	invokespecial	"System/Decimal" "<init>__iiiB" "(IIIZI)"
	putstatic	valuetype ['.library']'System'.'Decimal' 'Test'::'dc5'
	new	"System/Decimal"
	dup
	bipush	123
	iconst_0
	iconst_0
	iconst_0
	iconst_3
	invokespecial	"System/Decimal" "<init>__iiiB" "(IIIZI)"
	putstatic	valuetype ['.library']'System'.'Decimal' 'Test'::'dc6'
	new	"System/Decimal"
	dup
	ldc	int32(-1733296128)
	ldc	int32(-926269490)
	sipush	666
	iconst_0
	iconst_0
	invokespecial	"System/Decimal" "<init>__iiiB" "(IIIZI)"
	putstatic	valuetype ['.library']'System'.'Decimal' 'Test'::'dc7'
	new	"System/Decimal"
	dup
	ldc	int32(1870659584)
	ldc	int32(1443446532)
	ldc	int32(666784)
	iconst_0
	iconst_0
	invokespecial	"System/Decimal" "<init>__iiiB" "(IIIZI)"
	putstatic	valuetype ['.library']'System'.'Decimal' 'Test'::'dc8'
	new	"System/Decimal"
	dup
	ldc	int32(1870659584)
	ldc	int32(1443446532)
	ldc	int32(666784)
	iconst_0
	iconst_0
	invokespecial	"System/Decimal" "<init>__iiiB" "(IIIZI)"
	putstatic	valuetype ['.library']'System'.'Decimal' 'Test'::'dc9'
	new	"System/Decimal"
	dup
	bipush	123
	iconst_0
	iconst_0
	iconst_0
	bipush	23
	invokespecial	"System/Decimal" "<init>__iiiB" "(IIIZI)"
	putstatic	valuetype ['.library']'System'.'Decimal' 'Test'::'dc10'
	bipush	123
	invokestatic	"System/Decimal" "op_Implicit__iV" "(I)LSystem/Decimal;"
	putstatic	valuetype ['.library']'System'.'Decimal' 'Test'::'dc11'
	new	"System/Decimal"
	dup
	iconst_m1
	iconst_m1
	iconst_m1
	iconst_0
	iconst_0
	invokespecial	"System/Decimal" "<init>__iiiB" "(IIIZI)"
	putstatic	valuetype ['.library']'System'.'Decimal' 'Test'::'dc12'
	new	"System/Decimal"
	dup
	iconst_m1
	iconst_m1
	iconst_m1
	iconst_1
	iconst_0
	invokespecial	"System/Decimal" "<init>__iiiB" "(IIIZI)"
	putstatic	valuetype ['.library']'System'.'Decimal' 'Test'::'dc13'
	new	"System/Decimal"
	dup
	iconst_m1
	iconst_m1
	iconst_m1
	iconst_1
	bipush	20
	invokespecial	"System/Decimal" "<init>__iiiB" "(IIIZI)"
	putstatic	valuetype ['.library']'System'.'Decimal' 'Test'::'dc14'
	new	"System/Decimal"
	dup
	iconst_m1
	iconst_m1
	iconst_m1
	iconst_1
	bipush	28
	invokespecial	"System/Decimal" "<init>__iiiB" "(IIIZI)"
	putstatic	valuetype ['.library']'System'.'Decimal' 'Test'::'dc15'
	new	"System/Decimal"
	dup
	ldc	int32(-1717986918)
	ldc	int32(-1717986919)
	ldc	int32(429496729)
	iconst_1
	bipush	28
	invokespecial	"System/Decimal" "<init>__iiiB" "(IIIZI)"
	putstatic	valuetype ['.library']'System'.'Decimal' 'Test'::'dc16'
	new	"System/Decimal"
	dup
	iconst_1
	iconst_0
	iconst_0
	iconst_0
	bipush	28
	invokespecial	"System/Decimal" "<init>__iiiB" "(IIIZI)"
	putstatic	valuetype ['.library']'System'.'Decimal' 'Test'::'dc17'
	iconst_0
	invokestatic	"System/Decimal" "op_Implicit__iV" "(I)LSystem/Decimal;"
	putstatic	valuetype ['.library']'System'.'Decimal' 'Test'::'dc18'
	return
	.locals 0
	.maxstack 7
} // method .cctor
} // class Test
